윤준석/윤준석.sql

```diff
@@ -0,0 +1,118 @@
+/* 
+과제 A:
+주어진 요구사항에 맞게 SQL 질의문을 만드세요
+강의 자료로 주어진 테이블을 통해 만드시면 됩니다.
+ */
+-- 1. 전공 이름이 '소프트웨어공학전공'인 남학생을 모두 조회하세요
+SELECT s.*
+FROM student s 
+JOIN major m ON s.major_id = m.major_id
+WHERE m.major_name = '소프트웨어공학전공' AND s.gender = 'male';
+
+-- 2. 30000원이 넘는 책의 이름과 가격을 조회하세요(가격에 대해 내림차순으로 정렬해야 합니다)
+SELECT b.title, b.price
+FROM book b
+WHERE b.price > 30000
+ORDER BY price DESC;
+
+-- 3. 모든 학생의 수를 조회하세요
+SELECT COUNT(*) AS total_students
+FROM student;
+
+-- 4. 저자 이름이 '김영한' 혹은 '이일민'인 책의 개수를 조회하세요
+SELECT b.author, COUNT(*)
+FROM book b
+WHERE b.author = '김영한' OR b.author = '이일민'
+GROUP BY b.author;
+
+/*
+과제 B:
+1. 자신만의 데이터베이스를 만드세요.
+	- 주제 무관(2주차 과제에 사용한 주제를 재사용해도 상관 없습니다)
+    - 한 주제에 대해 데이터베이스를 구성하되, 테이블을 2개 이상 사용해야 합니다(반드시 기본키-외래키로 테이블을 연결할 필요는 없습니다)
+	- 예시) 학교 데이터베이스 구성 -> 학생 테이블, 전공 테이블 사용
+2. 1번에서 만든 데이터베이스를 통해 CRUD 질의문을 만드세요
+	- Create, Read, Update, Delete에 각각 1개 이상의 질의문을 만들어야 합니다
+    - Read, Update, Delete는 기본키를 이용한 질의문을 사용해도 좋고, 본인만의 방식으로 자유롭게 만드셔도 좋습니다.
+*/
+-- 자신만의 데이터베이스 만들기(테이블 2개 이상 생성)
+CREATE TABLE jejuair (
+jejuair_id int PRIMARY KEY,
+jejuair_name varchar(50),
+jejuair_passportnumber varchar(50)
+);
+/* 제주항공 고객관리 데이터베이스 테이블 생성 */
+
+CREATE TABLE FL7C131 (
+FL7C131_id int PRIMARY KEY,
+FL7C131_name varchar(50),
+FL7C131_passportnumber varchar(50),
+FL7C131_price int
+);
+/* 김포에서 제주로가는 제주항공 편명7C131 에 대한 탑승객 정보 테이블 생성 */
```

Comment:
작성하신 테이블을 보면, 고객의 정보(이름, 항공권 번호)가 양쪽 테이블에 중복되어 저장되고 있는 것 같네요.
이 경우, 만일 어떤 고객의 이름을 변경하고 싶다면 두 테이블 모두에 `UPDATE` 쿼리를 날려야겠죠?
만약 데이터를 변경하다가 실수로 `jejuair` 테이블만 수정하고 `FL7C131` 테이블을 수정하지 않으면 어떻게 될까요!
고객 정보에 대한 불일치가 발생하겠죠?

관계형 데이터베이스에서는, 이런 상황을 방지하고자 고객에 대한 정보를 한 테이블에 몰아서 저장합니다.
만약 위의 `FL7C131`처럼 고객에 대한 정보가 필요한 테이블이 있다면, 고객 테이블의 기본키를 참조하는 외래키를 만들어서 연결하면 되겠죠?
그렇게 되면 고객에 대한 정보를 수정할 일이 있어도, 다른 테이블은 일절 신경쓰지 않고 고객 테이블만 수정하면 되겠네요!

실세계의 정보를 DB로 옮겨 저장하는 것은 생각보다 쉽지 않아요.
직접 DB를 만들기 위해선 공부할게 많지만, 기본키와 외래키에 대해 숙달하는 것이 우선적으로 이루어져야한다고 생각해요.
직접 테이블을 구상하면서 외래키를 통해 각 테이블을 연결하는 연습을 하다 보면, 데이터를 어떻게 저장해야 하는지에 대해서도 감이 잡힐거에요.
1~2시간만에 제대로 설명하기 어려운 내용이라 스터디 세션에선 다루지 않았지만, 만약 관심이 있다면 저에게 질문하셔도 좋아요. :)


```suggestion
-- 자신만의 데이터베이스 만들기(테이블 2개 이상 생성)
CREATE TABLE jejuair (
jejuair_id int PRIMARY KEY,
jejuair_name varchar(50),
jejuair_passportnumber varchar(50)
);
/* 제주항공 고객관리 데이터베이스 테이블 생성 */

CREATE TABLE FL7C131 (
FL7C131_id int PRIMARY KEY,
FL7C131_price int,
jejuair_id int, /* 외래키를 통해 고객의 정보를 가져올 수 있게 함 */
constraint fk_jejuair foreign key (jejuair_id) references jejuair(jejuair_id)
);
/* 김포에서 제주로가는 제주항공 편명7C131 에 대한 탑승객 정보 테이블 생성 */
```

Reply:
기본키와 외래키에 대한 정의를 확실하게 알고 테이블을 작성하는 습관이 중요한 것 같습니다 감사합니다.
그러면 항공권 예약 / 항공사 고객정보 의 관점에서 본다면 항공사 고객정보에서 이름, 여권번호, 국적, 회원번호 등을 보유하고 있고
항공권 예약에서는 여권번호만 공통으로 구성하여 고객정보의 외래키로 참조하도록 구성하고, 예약한 좌석정보, 티켓 구입 가격, 기내식 선택 정보 등을 넣는게 더 낫겠군요.

Reply:
상세한 피드백 감사드립니다!